R/AlgorithmHandler.r

```diff
@@ -0,0 +1,95 @@
+                                                 base64enc::base64decode(input$data)
+                                               } else {
+                                                 input$data
+                                               }
+                                             }
+                                             getResponseObject_ <- function(output) {
+                                               if (typeof(output) == "raw") {
+                                                 list(result=base64enc::base64encode(output), metadata=list(content_type="binary"))
+                                               } else if (is.character(output) & length(output) == 1) {
+                                                 list(result=output, metadata=list(content_type="text"))
+                                               } else {
+                                                 list(result=output, metadata=list(content_type="json"))
+                                               }
+                                             }
+                                             getResponseAsJsonString_ <- function(output) {
+                                               tryCatch({
+                                                 rjson::toJSON(output)
+                                               },
+                                               error = function(e) {
+                                                 print(paste0("Error in getResponse: ", e))
+                                                 rjson::toJSON(list(error=list(message=toString(e),
+                                                                               stacktrace="pipe.r:getResponseAsJsonString", error_type="AlgorithmError")))
+                                               })
+                                             }
+                                            runLoad_ <- function(){
+                                              state <- onLoadMethod()
+                                              print("PIPE_INIT_COMPLETE")
+                                              flush.console()
+                                              state
+                                            }
+                                            outputFile <- file("/tmp/algoout")
+                                            inputFile <- file(pipe_name)
+                                            open(inputFile)
+                                            result <- tryCatch({
+                                            stage <- "loading"
+                                            state <- runLoad_()
+                                            list(state=state)
+                                            },
+                                            error = function(e){
+                                              message <- toString(e)
+                                              formatted <- gsub("\n", "", message)
```

Comment:
this is because the output is escaped, the `\n` converts into a `\\n` which means we end up having a bit of garbage at the end. I could be wrong but this seems to be what toString() does with exception messages.